File: src/data/class_label/abstract_class_label.py
from abc import ABC, abstractmethod


class AbstractClassLabel:
    _class_label = None
    __is_rejected = False

    def __init__(self, class_label):
        self._class_label = class_label

    def get_class_label_value(self):
        if self._class_label is None:
            raise Exception('Class label not defined')
        return self._class_label

    def set_class_label_value(self, class_label):
        self._class_label = class_label

    def is_rejected(self):
        return self.__is_rejected

    def set_rejected(self):
        self.__is_rejected = True

File: src/mofgbmlpy/core/data/class_label/abstract_class_label.cpp
#include "abstract_class_label.hpp"

AbstractClassLabel::AbstractClassLabel() : is_rejected_flag(false) {
}

bool AbstractClassLabel::is_rejected() const {
    return is_rejected_flag;
}

void AbstractClassLabel::set_rejected() {
    is_rejected_flag = true;
}

std::string AbstractClassLabel::to_string() const
{
    return static_cast<std::string>(*this);
}
